src/cim/components/ClusterDeployment/ClusterDeploymentHostsDiscoveryStep.tsx

```diff
@@ -12,65 +17,332 @@ import {
+  onSaveAgent: ClusterDeploymentHostsDiscoveryStepProps['onSaveAgent'];
+  onEditRole: ClusterDeploymentHostsDiscoveryStepProps['onEditRole'];
+  onSaveBMH: ClusterDeploymentHostsDiscoveryStepProps['onSaveBMH'];
+  onSaveISOParams: ClusterDeploymentHostsDiscoveryStepProps['onSaveISOParams'];
+  onFormSaveError: ClusterDeploymentHostsDiscoveryStepProps['onFormSaveError'];
+  fetchSecret: ClusterDeploymentHostsDiscoveryStepProps['fetchSecret'];
+  fetchNMState: ClusterDeploymentHostsDiscoveryStepProps['fetchNMState'];
+  onChangeBMHHostname: ClusterDeploymentHostsDiscoveryStepProps['onChangeBMHHostname'];
+  onApproveAgent: ClusterDeploymentHostsDiscoveryStepProps['onApproveAgent'];
+  onDeleteHost: ClusterDeploymentHostsDiscoveryStepProps['onDeleteHost'];
+  getClusterDeploymentLink: ClusterDeploymentHostsDiscoveryStepProps['getClusterDeploymentLink'];
+};
+
+const HostDiscoveryForm: React.FC<HostDiscoveryFormProps> = ({
   agentClusterInstall,
   agents: allAgents,
   infraEnv,
-  ...restProps
+  clusterDeployment,
+  bareMetalHosts,
+  onCreateBMH,
+  onSaveAgent,
+  onEditRole,
+  onSaveBMH,
+  onSaveISOParams,
+  onFormSaveError,
+  fetchSecret,
+  fetchNMState,
+  onChangeBMHHostname,
+  onApproveAgent,
+  onDeleteHost,
+  getClusterDeploymentLink,
+  ...rest
 }) => {
-  const { addAlert } = useAlerts();
   const { setCurrentStepId } = React.useContext(ClusterDeploymentWizardContext);
-
+  const [showClusterErrors, setShowClusterErrors] = React.useState(false);
+  const {
+    values,
+    isValid,
+    isValidating,
+    isSubmitting,
+    touched,
+    errors,
+    validateForm,
+    setTouched,
+    submitForm,
+  } = useFormikContext<ClusterDeploymentHostsDiscoveryValues>();
+  const [nextRequested, setNextRequested] = React.useState(false);
+  const [showFormErrors, setShowFormErrors] = React.useState(false);
   const infraEnvAgents = React.useMemo(
     () => allAgents.filter((a) => isAgentOfInfraEnv(infraEnv, a)),
     [allAgents, infraEnv],
   );
+  const selectedAgents = getSelectedAgents(allAgents, values);
+  const usedHostnames = React.useMemo(() => getAgentsHostsNames(allAgents), [allAgents]);
+  const onNext = async () => {
+    if (!showFormErrors) {
+      setShowFormErrors(true);
+      const errors = await validateForm();
+      setTouched(
+        Object.keys(errors).reduce((acc, curr) => {
+          acc[curr] = true;
+          return acc;
+        }, {}),
+      );
+      if (Object.keys(errors).length) {
+        return;
+      }
+    }
+    await submitForm();
+    setNextRequested(true);
+  };
+
+  React.useEffect(() => {
+    setNextRequested(false);
+    setShowClusterErrors(false);
+  }, [values.selectedHostIds]);
 
-  const [initialValues, validationSchema] = React.useMemo(
-    () => [
-      getInitialValues({ agents: infraEnvAgents, agentClusterInstall }),
-      getValidationSchema(),
-    ],
-    [], // eslint-disable-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
```

Comment:
while we wait for the agents to be bound to the cluster, we need to add some indication to the user about what is going on. Same as https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/cim/components/ClusterDeployment/ClusterDeploymentHostSelectionStep.tsx#L234-L239

Reply:
done!